flink-libraries/flink-table/src/main/scala/org/apache/flink/table/plan/schema/FlinkTableFunctionImpl.scala

```diff
@@ -36,10 +36,8 @@ import org.apache.flink.table.calcite.FlinkTypeFactory
 class FlinkTableFunctionImpl[T](
     val typeInfo: TypeInformation[T],
     val fieldIndexes: Array[Int],
-    val fieldNames: Array[String],
-    val evalMethod: Method)
-  extends ReflectiveFunctionBase(evalMethod)
```

Comment:
I suggest keep extends `ReflectiveFunctionBase` for keep consistent with calcite.

Reply:
In the first implementation of UDTF, we use `ReflectiveFunctionBase` to infer the operand types. But with the new approach in this PR, we customize a `SqlOperandTypeChecker` to check operands which means we don't need the `ReflectiveFunctionBase` anymore. If we extends `ReflectiveFunctionBase` we have to register every `eval` method as a UDTF. But with the new approach in this PR, we only need to register once.

Reply:
@wuchong I got you meaning, I'am find with your approach. And in order to find a better way, @fhueske and @twalthr I am appreciate if  you can give us some advices?

Reply:
I think we don't need `ReflectiveFunctionBase` anymore. The logic is implemented by us now.